src/runtime/agas/server/route.cpp

```diff
@@ -27,17 +27,17 @@
 #include <utility>
 #include <vector>
 
-
-namespace {
+namespace hpx { namespace detail {
     void update_agas_cache(hpx::naming::gid_type const& gid,
         hpx::naming::address const& addr, std::uint64_t count,
         std::uint64_t offset)
     {
-        hpx::naming::get_agas_client().update_cache_entry(gid, addr, count, offset);
+        hpx::naming::get_agas_client().update_cache_entry(
+            gid, addr, count, offset);
     }
-}
+}}    // namespace hpx::detail
```

Comment:
Was there a particular reason for keeping this in the anonymous namespace? Without this change I get
```
/home/simbergm/src/hpx/src/runtime/agas/server/route.cpp:40:40: warning: 'update_agas_cache_action' has a base 'hpx::actions::action<void (*)(c
onst hpx::naming::gid_type&, const hpx::naming::address&, long unsigned int, long unsigned int), {anonymous}::update_agas_cache, update_agas_ca
che_action>' whose type uses the anonymous namespace [-Wsubobject-linkage]                                                                     
   40 | HPX_PLAIN_ACTION_ID(update_agas_cache, update_agas_cache_action,                                                                       
      |                                        ^~~~~~~~~~~~~~~~~~~~~~~~                                                                        
../../libs/preprocessor/include/hpx/preprocessor/expand.hpp:23:26: note: in definition of macro 'HPX_PP_EXPAND'                                
   23 | #define HPX_PP_EXPAND(X) X                                                                                                             
      |                          ^                                     
../../libs/preprocessor/include/hpx/preprocessor/cat.hpp:36:28: note: in expansion of macro 'HPX_DEFINE_PLAIN_ACTION_2'                        
   36 | #define HPX_PP_CAT_I(a, b) a##b                                
      |                            ^                                   
../../hpx/runtime/actions/plain_action.hpp:187:5: note: in expansion of macro 'HPX_DEFINE_PLAIN_ACTION_'                                       
  187 |     HPX_DEFINE_PLAIN_ACTION_(__VA_ARGS__)                                     \                                                        
      |     ^~~~~~~~~~~~~~~~~~~~~~~~                                   
../../hpx/runtime/actions/plain_action.hpp:342:5: note: in expansion of macro 'HPX_DEFINE_PLAIN_ACTION'                                        
  342 |     HPX_DEFINE_PLAIN_ACTION(func, name);                                      \
      |     ^~~~~~~~~~~~~~~~~~~~~~~                                    
/home/simbergm/src/hpx/src/runtime/agas/server/route.cpp:40:1: note: in expansion of macro 'HPX_PLAIN_ACTION_ID'                               
   40 | HPX_PLAIN_ACTION_ID(update_agas_cache, update_agas_cache_action,                                                                       
      | ^~~~~~~~~~~~~~~~~~~                                            
```
but I'm not sure if I've messed something up by doing this (it compiles, but I haven't tested it).

I can also instead of this change mark the file to be compiled alone even if `UNITY_BUILD` is enabled.

Reply:
There is no particular reason.